### PR TITLE
feat/junit engine

### PR DESCRIPTION
**Pull request by boddissattva**
First attempt of JUnit5 engine in Koltin for Kotlin....

### CODE REVIEW

**Review by nbrouand — CHANGES_REQUESTED** [review]

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/launcher/ExecutionUtils.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            securityProperties.trustStore?.ifEmpty { null },
+            securityProperties.trustStorePassword?.ifEmpty { null },
+            securityProperties.keyStore?.ifEmpty { null },
+            securityProperties.keyStorePassword?.ifEmpty { null },
+            securityProperties.privateKey?.ifEmpty { null }
+        )
+    }
+}
+
+object ReportWriter {
```
Comment (bessonm) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/launcher/ExecutionUtils.kt:99]: ReportWriter is used by the Launcher and the JUnit engine.
Being buried here in launcher package in an "util" class doesn't feel right

**Review by bessonm — CHANGES_REQUESTED** [review]
I'm ok with most of the code except its file organization.
There are 2 classes "ExecutionUtils" one plural, one singular.

ReportWriter and ExecutionService are shared between Launcher and Engine.

Overall understanding and reading is not the best.
I feel like Kotlin is to easy to clutter.

We should also remove all wildcard imports, it doesn't help to improve reading since we cannot get a glance at dependencies.

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyExecutionContexts.kt [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    }
+
+    private fun unregisterRxBus() {
+        startScenarioDisposable.dispose()
+        beginStepDisposable.dispose()
+        endStepDisposable.dispose()
+        endScenarioDisposable.dispose()
+    }
+
+    private fun executionStatus(): Status {
```
Comment (bessonm) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyExecutionContexts.kt:157]: unused method, still relevant ?

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/execution/report/SiteGenerator.kt [hunk 3]
```diff
@@ -1,5 +1,6 @@
-package com.chutneytesting.kotlin.launcher
+package com.chutneytesting.kotlin.execution.report
```
Comment (bessonm) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/execution/report/SiteGenerator.kt:1]: l.70 there is commented code we can delete

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyExecutionContexts.kt [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+        }
+    }
+
+    fun endExecution() {
+        endExecutionLatch.await()
+        unregisterRxBus()
+
+        request.engineExecutionListener.executionFinished(
+            request.rootTestDescriptor,
+            TestExecutionResult.successful() // global status could be executionStatus() but brings some integration problems with ide
```
Comment (bessonm) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyExecutionContexts.kt:97]: I don't understand those lines. Is it really used for something ? I don't see any different behavior when removing them

**Comment by nbrouand** [comment]
I would like to add in test a junit properties file with all configurable properties to be able to play easily with in our projet

**Comment by nbrouand** [comment]
Comment for futur upgrade : 
 - Manage "Rerun failed test"
 - Click in left menu redirect to the scenario, or the step

**Review by nbrouand — APPROVED** [review]

**Review by bessonm — APPROVED** [review]

**Pull request merged**